src/lib.rs

```diff
@@ -164,6 +164,26 @@
 //! }
 //! ```
 //!
+//!```compile_fail
+//! use evmap::ReadHandle;
+//!
+//! fn is_sync<T: Sync>() {
+//!   // dummy function just used for its parameterized type bound
+//! }
+//!
+//! is_sync::<ReadHandle<u64, u64>>()
+//!```
+//!
+//!```
+//! use evmap::ReadHandle;
+//!
+//! fn is_send<T: Send>() {
+//!   // dummy function just used for its parameterized type bound
+//! }
+//!
+//! is_send::<ReadHandle<u64, u64>>()
+//!```
+//!
```

Comment:
I refactored to put this in a doc test. My only concern is that it doesn't test exclusively that the trait is inherited so can pass if any line of code doesn't compile.

I refactored to this because it seemed that:
a) It was better supported in the rust ecosystem since the language itself uses this doctest macro
b) I couldn't get the static assertions macro to fail in ways that I would expect and I did not fully grok how it worked. Also admittedly could make more effort on that front.

Reply:
Yup, this looks like a good approach!